wordpress-mobile/WordPress-iOS
# Fix Core Data concurrency issues in `AccountService`

I noticed some Core Data concurrency issues on accessing `WPAccount` from a wrong thread.

I don't see a huge risk associated with these changes. The `updateDefaultBlogIfNeeded` function change is only a signature change, the code is still executed the way it used to be. The real logic of `setupAppExtensionsWithDefaultAccount` is now forced to run in the main thread, which is where it used to run too.

## Test instructions

A couple of features to verify:
- Log in and log out of the app using a WP.com account.
- "Save as Draft" share extension (i.e. from Safari).

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What's described in the "Test instructions"

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:
